test/e2e/utils/text_utils.go

```diff
@@ -0,0 +1,55 @@
+package utils
+
+import (
+	"bufio"
+	"regexp"
+	"strings"
+)
+
+// StringLines split string
+func StringLines(s string) []string {
+	return strings.Split(s, "\n")
+}
+
+// StringHead return first line of a string with multiple lines
+func StringHead(s string) string {
+	sl := strings.SplitN(s, "\n", 1)
+	if len(sl) > 0 {
+		return sl[0]
+	}
+	return ""
+}
+
+// StringExtractLineMatching Extract the first line matching the given regexp condition
+func StringExtractLineMatching(s string, matchRexp string) string {
+
+	r := regexp.MustCompile(matchRexp)
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	res := ""
+	for scanner.Scan() {
+		line := scanner.Text()
+		if r.MatchString(line) {
+			res = line
+			break
+		}
+	}
+	return res
+}
+
+// StringExtractLineFieldsMatching Extract fields which line field matches a given regexp condition
+func StringExtractLineFieldsMatching(s string, fieldIndex int, matchRexp string) []string {
```

Comment:
The fact that a `StringExtractLineFieldsMatching` function even exists in what should be a simple E2E test PR unfortunately tells me this is much too complicated and needs a drastic overhaul before I am willing to either accept the PR or the responsibility of maintaining this code going forward.

Reply:
Per our conversation in Slack, I am resolving this and the other comments.  Thank you @jrangelramos for taking the time to discuss most of these comments, and for creating the associated design document in Google Docs.  I am looking forward to seeing the final version